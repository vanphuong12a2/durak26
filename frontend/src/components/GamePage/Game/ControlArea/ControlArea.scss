.control-area {
  text-align: right;

  a {
    margin-right: 5px;

    &:last-child {
      margin-right: 0;
    }
  }
}
