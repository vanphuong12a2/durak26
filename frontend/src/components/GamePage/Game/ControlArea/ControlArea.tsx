import React from 'react';
import './ControlArea.scss';
import {RouteComponentProps} from 'react-router';

interface Props {
  playing: boolean
  currentPlayerId?: string
  exitGameFunction: () => void
  serveGameFunction: () => void
}

const ControlArea = (props: Props & RouteComponentProps) => {

  const exitGameHandler = () => {
    props.exitGameFunction();
    props.history.push('/');
  };

  const serveGameButtonText = props.playing ? 'Restart game' : 'Start game';

  return (
    <div className='control-area'>
      {props.currentPlayerId && <a id='start-game-btn'
                                   className='waves-effect waves-light btn-small'
                                   onClick={props.serveGameFunction}>
        {serveGameButtonText}
      </a>}
      <a id='exit-game-btn'
         className='waves-effect waves-light red lighten-2 btn-small'
         onClick={exitGameHandler}>
        Exit
      </a>
    </div>
  );
};

export default ControlArea;
