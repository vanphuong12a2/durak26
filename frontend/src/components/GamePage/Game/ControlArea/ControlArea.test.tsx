import React from 'react';
import {create} from 'react-test-renderer';
import ControlArea from './ControlArea';

describe('<ControlArea />', () => {

  describe('Snapshots', () => {
    it('renders correctly', () => {
      const app = create(<ControlArea />);
      expect(app.toJSON()).toMatchSnapshot();
    });
  });

});
