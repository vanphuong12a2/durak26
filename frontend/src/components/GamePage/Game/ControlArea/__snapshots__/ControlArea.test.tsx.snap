// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlArea /> Snapshots renders correctly 1`] = `
<div
  className="control-area"
>
  <a
    className="waves-effect waves-light red lighten-2 btn-small"
    id="exit-game-btn"
    onClick={[Function]}
  >
    Exit
  </a>
</div>
`;

exports[`<ControlArea /> Snapshots renders correctly without Start game button 1`] = `
<div
  className="control-area"
>
  <a
    className="waves-effect waves-light btn-small"
    id="start-game-btn"
    onClick={[MockFunction]}
  >
    Start game
  </a>
  <a
    className="waves-effect waves-light red lighten-2 btn-small"
    id="exit-game-btn"
    onClick={[Function]}
  >
    Exit
  </a>
</div>
`;
